src/expr/src/expr/mod.rs

```diff
@@ -56,6 +56,14 @@ pub type ExpressionRef = Arc<dyn Expression>;
 pub trait Expression: std::fmt::Debug + Sync + Send {
     fn return_type(&self) -> DataType;
 
+    fn wrapping_eval(&self, input: &DataChunk) -> Result<ArrayRef> {
+        let res = self.eval(input)?;
+
+        assert_eq!(res.len(), input.capacity());
```

Comment:
off topic: another useful check would be that array variant matches expected return type.